tests/functional_tests/test_query_builder.py

```diff
@@ -4,6 +4,95 @@
 import io
 import pandas as pd
 from datenguidepy import Field, Query
+from datenguidepy.query_execution import FieldMetaDict, TypeMetaData
+
+
+@pytest.fixture
+def mocked_meta_data(monkeypatch):
+    def mock_enum_placeholder(self):
+        return "MOCKED ENUM VALUES"
+
+    def mock_get_arguments(self):
+        mocked_args = {
+            "year": ("LIST", None, "SCALAR", "Int"),
+            "statistics": ("LIST", None, "ENUM", "BEV001Statistics"),
+            "ALTMT1": ("LIST", None, "ENUM", "ALTMT1"),
+        }
+        return mocked_args
+
+    def mock_get_description(self, *args):
+        return "Lebend Geborene"
+
+    def mock_field_meta_data(self):
+
+        type_kind = "kind='OBJECT'"
+        field_meta = {
+            "id": FieldMetaDict(
+                {
+                    "name": "id",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "String",
+                        "description": "The `String` scalar...",
+                    },
+                    "description": "Interne eindeutige ID",
+                    "args": [],
+                }
+            ),
+            "year": FieldMetaDict(
+                {
+                    "name": "year",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "Int",
+                        "description": "The `Int` scalar...",
+                    },
+                    "description": "Jahr des Stichtages",
+                    "args": [],
+                }
+            ),
+            "value": FieldMetaDict(
+                {
+                    "name": "value",
+                    "type": {
+                        "ofType": None,
+                        "kind": "SCALAR",
+                        "name": "Float",
+                        "description": "The `Float` scalar...",
+                    },
+                    "description": "Wert",
+                    "args": [],
+                }
+            ),
+        }
+        enum_vals = None
+        return Field._no_none_values(
+            self._fields_info_formatter,
+            TypeMetaData(type_kind, field_meta, enum_vals),
+            "fields",
+        )
+
+    monkeypatch.setattr(Field, "_get_description", mock_get_description)
+    monkeypatch.setattr(FieldMetaDict, "get_arguments", mock_get_arguments)
```

Comment:
Why is the FieldMetaDict class mocked? Wouldn't it be more natural to mock the graphql meta data provider which returns the FieldMetaDict instance, to return the right values, or is there an issue with this. This way we take a container class that is not responsible for its content and mock out a method which therefore becomes responsible for content.

Reply:
same as above - 
and my though was: if mainly the query builder is supposed to be tested, then there is not that much of a difference wether the FieldMetaDict is mocked or the Graphql Meta Data Provider?

Reply:
There is no issue from the functionality side. There is a good practice that I try to adhere to that tries to split logic for data containers and business logic. This mock mixes the two a little bit and that is why I don't like it. But I actually find it more confusing in the case of the unit tests where it is also mocked.